tests/MongoDB.Driver.Core.Tests/Core/Authentication/ScramSha256AuthenticatorTests.cs

```diff
@@ -261,5 +261,68 @@ public void Constructor_should_throw_an_ArgumentNullException_when_credential_is
                            conversationId: 1, " + 
                 $"         payload: new BinData(0, \"{ToUtf8Base64(_clientRequest2)}\")}}}}");
         }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Authenticate_should_use_cache(
+            [Values(false, true)] bool async)
+        {
+            var randomStringGenerator = new ConstantRandomStringGenerator(_clientNonce);
+            var subject = new ScramSha256Authenticator(__credential, randomStringGenerator);
+
+            var saslStartReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                @"{conversationId: 1," +
+                $" payload: BinData(0,\"{ToUtf8Base64(_serverResponse1)}\")," +
+                @" done: false,
+                   ok: 1}"));
+            var saslContinueReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                @"{conversationId: 1," +
+                $" payload: BinData(0,\"{ToUtf8Base64(_serverReponse2)}\")," +
+                @" done: true,
+                   ok: 1}"));
+
+            var connection = new MockConnection(__serverId);
+            connection.EnqueueReplyMessage(saslStartReply);
+            connection.EnqueueReplyMessage(saslContinueReply);
+
+            Action act;
+            if (async)
+            {
+                act = () => subject.AuthenticateAsync(connection, __description, CancellationToken.None).GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                act = () => subject.Authenticate(connection, __description, CancellationToken.None);
+            }
+
+            var exception = Record.Exception(act);
+            exception.Should().BeNull();
```

Comment:
No need. If an unexpected exception were thrown the test will already fail.

Reply:
Done